similarity-search/src/main/java/org/phenotips/similarity/internal/SolrSimilarPatientsFinder.java

```diff
@@ -155,16 +166,15 @@ private SolrQuery generateQuery(Patient referencePatient, boolean prototypes)
     {
         SolrQuery query = new SolrQuery();
         StringBuilder q = new StringBuilder();
-        // FIXME This is a very basic implementation, to be revisited
-        for (Feature phenotype : referencePatient.getFeatures()) {
-            if (StringUtils.isNotBlank(phenotype.getId())) {
-                q.append(phenotype.getType() + ":" + ClientUtils.escapeQueryChars(phenotype.getId()) + " ");
-            }
+        // Whitespace-delimiter terms querying the extended phenotype field
+        Collection<String> termIds = getPresentPhenotypeTerms(referencePatient);
+        if (!termIds.isEmpty()) {
+            q.append(" extended_phenotype: " + getQueryFromTerms(termIds));
```

Comment:
Hm, it actually depends on the query type. A plain lucene query does work like this, but not a dismax query.